tests/contrib/hooks/test_gcs_hook.py

```diff
@@ -602,21 +604,103 @@ def test_upload(self, mock_service):
         self.assertIsNone(response)
 
+    @mock.patch(GCS_STRING.format('GoogleCloudStorageHook.get_conn'))
+    def test_upload_string_strdata(self, mock_service):
+        test_bucket = 'test_bucket'
+        test_object = 'test_object'
+
+        upload_method = mock_service.return_value.bucket.return_value\
+            .blob.return_value.upload_from_string
+        upload_method.return_value = None
+
+        response = self.gcs_hook.upload(test_bucket,  # pylint:disable=assignment-from-no-return
+                                        test_object,
+                                        data=self.testdata_str)
+
+        self.assertIsNone(response)
+        upload_method.assert_called_once_with(
+            self.testdata_str,
+            content_type='text/plain'
+        )
+
+    @mock.patch(GCS_STRING.format('GoogleCloudStorageHook.get_conn'))
+    def test_upload_string_bytedata(self, mock_service):
+        test_bucket = 'test_bucket'
+        test_object = 'test_object'
+
+        upload_method = mock_service.return_value.bucket.return_value\
+            .blob.return_value.upload_from_string
+        upload_method.return_value = None
+
+        response = self.gcs_hook.upload(test_bucket,  # pylint:disable=assignment-from-no-return
+                                        test_object,
+                                        data=self.testdata_bytes)
+
+        self.assertIsNone(response)
+        upload_method.assert_called_once_with(
+            self.testdata_bytes,
+            content_type='text/plain'
+        )
+
+    @mock.patch(GCS_STRING.format('GoogleCloudStorageHook.get_conn'))
+    def test_upload_string_gzip(self, mock_service):
+        test_bucket = 'test_bucket'
+        test_object = 'test_object'
+
+        upload_method = mock_service.return_value.get_bucket.return_value\
+            .blob.return_value.upload_from_string
+        upload_method.return_value = None
+
+        response = self.gcs_hook.upload(test_bucket,  # pylint:disable=assignment-from-no-return
+                                        test_object,
+                                        data=self.testdata_str,
+                                        gzip=True)
+
+        self.assertIsNone(response)
+
+        response = self.gcs_hook.upload(test_bucket,  # pylint:disable=assignment-from-no-return
+                                        test_object,
+                                        data=self.testdata_bytes,
+                                        gzip=True)
+
+        self.assertIsNone(response)
+
+    @mock.patch(GCS_STRING.format('GoogleCloudStorageHook.get_conn'))
+    def test_upload_exceptions(self, mock_service):
+        test_bucket = 'test_bucket'
+        test_object = 'test_object'
+
+        upload_method = mock_service.return_value.get_bucket.return_value\
+            .blob.return_value.upload_from_string
+        upload_method.return_value = None
+
+        self.assertRaises(ValueError,
```

Comment:
It's not really nice way of using asserRaises. It is difficult to read. I think it should look similar to:

```
with self.assertRaises(ValueError) as cm:
    self.gcs_hook.upload(test_bucket,test_object)

the_exception = cm.exception
self.assertEqual("... Message here...", the_exception.message)
```